docs/How-To-Open-a-Homebrew-Pull-Request.md

```diff
@@ -104,24 +108,44 @@ To make changes on a new branch and submit it for review, create a GitHub pull r
 
 4. Make your changes. For formulae or casks, use `brew edit` or your favourite text editor, following all the guidelines in the [Formula Cookbook](Formula-Cookbook.md) or [Cask Cookbook](Cask-Cookbook.md).
    * If there's a `bottle do` block in the formula, don't remove or change it; we'll update it when we merge your PR.
+
 5. Test your changes by running the following, and ensure they all pass without issue. For changed formulae and casks, make sure you do the `brew audit` step after your changed formula/cask has been installed.
 
    ```sh
    brew tests
-   HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <CHANGED_FORMULA|CHANGED_CASK>
-   brew test <CHANGED_FORMULA|CHANGED_CASK>
-   brew audit --strict --online <CHANGED_FORMULA|CHANGED_CASK>
+   HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <FORMULA|CASK>
+   brew test <FORMULA|CASK>
+
+   brew style --fix <FORMULA|CASK>
+
+   brew audit --new <FORMULA|CASK>              # For new formula / cask
+   brew audit --strict --online <FORMULA|CASK>  # For existing formula / cask
    ```
 
 6. [Make a separate commit](Formula-Cookbook.md#commit) for each changed formula with `git add` and `git commit`.
-   * Please note that our preferred commit message format for simple version updates is "`<FORMULA_NAME> <NEW_VERSION>`", e.g. "`source-highlight 3.1.8`".
+   * Please note that for formula, our preferred commit message format for simple version updates is "`<FORMULA_NAME> <NEW_VERSION>`", e.g. "`source-highlight 3.1.8`".
+
 7. Upload your branch of new commits to your fork:
 
    ```sh
    git push --set-upstream <YOUR_USERNAME> <YOUR_BRANCH_NAME>
    ```
 
-8. Go to the relevant repository (e.g. <https://github.com/Homebrew/brew>, <https://github.com/Homebrew/homebrew-core>, etc.) and create a pull request to request review and merging of the commits from your pushed branch. Explain why the change is needed and, if fixing a bug, how to reproduce the bug. Make sure you have done each step in the checklist that appears in your new PR.
+   If you are using [GitHub two-factor authentication](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa) and have set your remote repository as HTTPS you will need to [set up a personal access token](https://docs.github.com/en/repositories/creating-and-managing-repositories/troubleshooting-cloning-errors#provide-an-access-token) and use that instead of your password.
+
+8. To create a pull request on GitHub for your changes, you can follow one of these methods:
```

Comment:
I wonder if we should be so explicit about detailing how to use git here; or if we could just link off the relevant page on GitHub's own help docs.

Reply:
If there's one as relevant as these steps: I'm open to exploring that.

Reply:
I haven't looked through all of this specifically yet, but this is the vague space of things I was thinking of:

- https://docs.github.com/en/pull-requests
- https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-and-deleting-branches-within-your-repository
- https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request
- https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests

I think it's fine for the Homebrew docs to have some quick copy/pasteable 'basic git' commands, and maybe some very slight explanation for it; but it doesn't seem like it should try and handhold users through every possible nuance of learning/using git/GitHub/Pull Requests/etc; which is where I think there is benefit in linking out to the relevant GitHub help docs where they can be way more verbose/explicit/hand holdy.

Reply:
I dunno. Reading this again: I think I'd rather we did keep these docs here. They are already written, correct, specific and helpful in a way GitHub's documentation is not.

Reply:
@MikeMcQuaid _nods_ fair enough. Do you think we need to keep the full "here are 10 different ways to do it via git" stuff? Or is the simpler stuff that existed before this PR pulled the extra bits over from the older cask section good enough?

I haven't read through this part for a while, but if it fits in with the content that's here well, would you be happy with some kind of short "if you want to know more about THIS_GITHUB_THING, you can read about it in their help docs [LINK]".

Generally when I write technical documentation I like to assume the person knows nothing, and make it as easy as possible to either give them the information, or point them to a canonical source where they can learn more themselves.

Reply:
Yes, I think it would be reasonable to slim this down to a single path and remove the alternatives.

Reply:
@MikeMcQuaid Can do.

Do you have an opinion on what I mentioned above RE: referencing GitHub's docs? (unless you also intended that by your above comment, but if so, it was unclear)

Reply:
> Do you have an opinion on what I mentioned above RE: referencing GitHub's docs? (unless you also intended that by your above comment, but if so, it was unclear)

I don't think there's any need. I'm happy to see this slimmed this down to a single path but I don't think GitHub's docs are necessary or helpful here, really.